var/spack/repos/builtin/packages/intel-parallel-studio/package.py

```diff
@@ -142,3 +154,122 @@ def install(self, spec, prefix):
 
         os.symlink(os.path.join(self.prefix.man, "common", "man1"),
                    os.path.join(self.prefix.man, "man1"))
+
+    def setup_environment(self, spack_env, run_env):
+        major_ver = self.version[1]
+
+        # Remove paths that were guessed but are incorrect for this package.
+        run_env.remove_path('LIBRARY_PATH',
+                            join_path(self.prefix, 'lib'))
+        run_env.remove_path('LD_LIBRARY_PATH',
+                            join_path(self.prefix, 'lib'))
+        run_env.remove_path('CPATH',
+                            join_path(self.prefix, 'include'))
+
+        # Add the default set of variables
+        run_env.prepend_path('LIBRARY_PATH',
+                             join_path(self.prefix, 'lib', 'intel64'))
+        run_env.prepend_path('LD_LIBRARY_PATH',
+                             join_path(self.prefix, 'lib', 'intel64'))
+        run_env.prepend_path('LIBRARY_PATH',
+                             join_path(self.prefix, 'tbb', 'lib',
+                                       'intel64', 'gcc4.4'))
+        run_env.prepend_path('LD_LIBRARY_PATH',
+                             join_path(self.prefix, 'tbb', 'lib',
+                                       'intel64', 'gcc4.4'))
+        run_env.prepend_path('CPATH',
+                             join_path(self.prefix, 'tbb', 'include'))
+        run_env.prepend_path('MIC_LIBRARY_PATH',
+                             join_path(self.prefix, 'lib', 'mic'))
+        run_env.prepend_path('MIC_LD_LIBRARY_PATH',
+                             join_path(self.prefix, 'lib', 'mic'))
+        run_env.prepend_path('MIC_LIBRARY_PATH',
+                             join_path(self.prefix, 'tbb', 'lib', 'mic'))
+        run_env.prepend_path('MIC_LD_LIBRARY_PATH',
+                             join_path(self.prefix, 'tbb', 'lib', 'mic'))
+
+        if self.spec.satisfies('+all'):
+            run_env.prepend_path('PATH',
+                                 join_path(self.prefix,
+                                           'debugger_{0}'.format(major_ver),
+                                           'gdb', 'intel64_mic', 'bin'))
```

Comment:
The intel sourcing scripts indicate it should be in the path and the intel64 directory was picked up elsewhere. However, the binaries in intel64_mic are also symlinked in the main bin link so the above can be removed.